madara-prover-rpc-server/src/services/prover.rs

```diff
@@ -39,10 +39,10 @@ fn format_execution_result(
 
 /// Formats the output of the prover subprocess into the server response.
 fn format_prover_result(
-    prover_result: Result<Proof, ProverError>,
+    prover_result: Result<(Proof, ProverWorkingDirectory), ProverError>,
```

Comment:
The change was directly related to changing various fns to return `Result<(Proof, PWD), _>` (previously `Result<Proof, _>`).

The fn was problematic for me before this, though, because it wants to be greedy with (consume) the `Result`, which gets messy when anything else needs to also consume the Result.

An idea for cleaning it up:

Make a `From` impl so that we can easily convert `ProverError` to `Status` -- that may be as simple as converting `from_prover_error()` into such an impl. This lets us use the `?` operator for the Err cases. Then `format_prover_result` can forget about its `Err` case and operate only on a `Proof`.